### ISSUE TITLE
Algolia website search does not work

### ISSUE THREAD

**Issue by callumlocke** [reduxjs/redux-toolkit#3816]
The docs site is generally good and well written, but the Algolia search engine seems to be useless. Even for simple searches, the ranking of results makes no sense, and the titles of the results are misleading.

As an example, try searching "createAsyncThunk". This is a core RTK library function that has a single dedicated, detailed documentation page titled simply `createAsyncThunk`. I can't think of an easier case for a search engine to surface the correct page. But that page is actually buried under about 12 spurious results. To make it worse, a couple of the spurious results (including the very top result) are promisingly titled, exactly, "createAsyncThunk" - but if you click through you just get taken to an anchor on some other page that mentions `createAsyncThunk`. You might try adding "api" to the search term, but that doesn't help at all, it seems to just make the results even more random. It's the exact same story if you search for something else like `createSlice`.

Unless others are finding it useful, I'd suggest removing the Algolia search from the docs site, as for me it has just been a confusing, frustrating timewaster while learning RTK.

**Comment by Dovakeidy** [comment]
I personally find it very useful. If I take the example of **createAsyncThunk**, the first results are exactly what I'm looking for.
The purple highlight shows that the exact word has been found, and the other links are generally directly related to the search. 

![Capture](https://github.com/reduxjs/redux-toolkit/assets/26869114/08e1a4cf-bcf8-4aed-a94e-438a89a376f1)

Searches are sorted by section (rtk, rtk-query .. ). So if you only search on **createAsync**, the most relevant result is in the 2nd section and not in the first results. However, with all the results displayed, I personally always find what I'm looking for.

![Capture1](https://github.com/reduxjs/redux-toolkit/assets/26869114/4c17c7cc-857e-4971-ae24-97cb4ac568dd)

Personally, I wouldn't recommend deleting it at all, even though I agree **it's not perfect**, it has helped me **more than once**!

**Comment by callumlocke** [comment]
I'm not sure what you mean when you say you 'would' be looking for the top result. The top result is the "Usage with TypeScript" page. My issue describes the search UX for someone who is looking for the [createAsyncThunk page](https://redux-toolkit.js.org/api/createAsyncThunk). My point is there is no way (with acceptable UX) to find that page through search. While the page does appear in the results, it is nowhere near the top, and there is no way for a user to identify which result it is (unless they are already intimately familiar with the site's navigational structure, and with the very particular way this structure is signposted in the search results).

**Comment by Dovakeidy** [comment]
You're right, I didn't notice that it wasn't the dedicated page. 

As I was saying, the search results are divided by section. To find the link to this page, you need to go down and take the first result in the "Reducers and Actions" section. I've often found the search useful, but sometimes you do have to look through the search results to find the best match. I have the impression that the problem of relevance lies rather in the order of the categories listed in the results, which doesn't necessarily correspond to the most relevant. 

I agree that this part should be improved, but I think it would be a mistake to remove the search function altogether, as it remains practical despite these concerns.

**Comment by callumlocke** [comment]
Fair enough, if it's providing value to some people then removing it isn't the right solution. If it can be improved by configuring the Algolia integration then that would be good, or by replacing it with something better, even if it's just a form that points to a third party search engine with a `site:` filter.

**Comment by markerikson** [comment]
We use Algolia because it's free for OSS projects.

If folks have better suggestions I'm interested, and even better if someone is willing to file PRs to improve things.  But messing with docs config is at the bottom of the priority list right now.

**Comment by markerikson** [comment]
Just switched all our docs sites over to using a local index generated during build, using Canary + Pagefind. Seems to produce better results overall, and the results list shows more hit details. Hopefully that helps!
